python/ml4ir/applications/ranking/model/metrics/helpers/metrics_helper.py

```diff
@@ -62,8 +139,20 @@ def get_grouped_stats(
         computed from the old and new ranks and aux labels generated
         by the model
     """
-    # Filter unclicked queries
-    df_clicked = df[df[label_col] == 1.0]
+    # adding "top_graded_relevance" to be the artificial click such that the record with the highest graded
+    # relevance is considered to be clicked nad will be used in MRR computation
+    artificial_click_col = "top_graded_relevance"
+    df = add_top_graded_relevance_record_column(df, label_col, artificial_click_col)
+    if np.array([Metric.NDCG in m for m in power_analysis_metrics]).any():
+        df = compute_ndcg(df, label_col, pred_col="ranking_score", new_col=RankingConstants.NEW_NDCG)
+
+        if old_ranking_score in df.columns:  # if the old model ranking score is available
+            df = compute_ndcg(df, label_col, pred_col=old_ranking_score, new_col=RankingConstants.OLD_NDCG)
```

Comment:
@mohazahran Can't we just approximate this based on 1/old rank? Otherwise you have to update a bunch of things in feature config

Reply:
but why do we need approx. if we have the real old ranking score?

Reply:
Why do you assume that in a public repo though?

Reply:
I don't see why this is a far fetched assumption for a public repo. Public datasets can have old scores as much as old rankings.